src/mlpack/methods/ann/layer/celu_impl.hpp

```diff
@@ -29,11 +30,55 @@ CELUType<InputType, OutputType>::CELUType(const double alpha) :
   }
 }
 
-template<typename InputType, typename OutputType>
-void CELUType<InputType, OutputType>::Forward(
-    const InputType& input, OutputType& output)
+template<typename MatType>
+CELUType<MatType>::CELUType(const CELUType& other) :
+    Layer<MatType>(other),
+    alpha(other.alpha)
 {
-  output = arma::ones<OutputType>(arma::size(input));
+    // Nothing to do.
+}
+
+template<typename MatType>
+CELUType<MatType>::CELUType(
+    CELUType&& other) :
+    Layer<MatType>(std::move(other)),
+    alpha(std::move(other.alpha))
+{
+    // Nothing to do.
+}
+
+template<typename MatType>
+CELUType<MatType>&
+CELUType<MatType>::operator=(const CELUType& other)
+{
+    if (&other != this)
+    {
+        Layer<MatType>::operator=(other);
+        alpha = other.alpha;
+    }
+
+    return *this;
+}
+
+template<typename MatType>
+CELUType<MatType>&
+CELUType<MatType>::operator=(CELUType&& other)
+{
+    if (&other != this)
+    {
+        Layer<MatType>::operator=(std::move(other));
+        alpha = std::move(other.alpha);
+    }
+
+    return *this;
+}
+
+template<typename MatType>
+void CELUType<MatType>::Forward(
+    const MatType& input, MatType& output)
+{
+  output = arma::ones<MatType>(arma::size(input));
+  derivative = arma::ones<MatType>(arma::size(input));
```

Comment:
Why do we set the values of `output` and `derivative` here?  `output` will already be the right size, and we overwrite all the values anyway.  So these two lines seem unnecessary.

Reply:
I did this to avoid errors in the cases that the user doesn't set the sizes of `output` and `derivative`. Should we still assume them to be the right size?

Reply:
Yeah, here the user will not call these functions directly, but instead they are called internally in the FFN and RNN code.  So assuming that things are already sized correctly is reasonable. :+1: